src/renderer/viz/expressions/ramp.js

```diff
@@ -325,6 +342,35 @@ export default class Ramp extends BaseExpression {
         gl.uniform1f(this._getBinding(program).keyMinLoc, (this.minKey));
         gl.uniform1f(this._getBinding(program).keyWidthLoc, (this.maxKey) - (this.minKey));
         drawMetadata.freeTexUnit++;
+
+        if (this.input.type === 'category' && this.input.isA(Property)) {
+            gl.activeTexture(gl.TEXTURE0 + drawMetadata.freeTexUnit);
+            if (this._translatedIds !== this._metadata.properties[this.input.name].categories.length) {
```

Comment:
Save `this._metadata.properties[this.input.name].categories.length` to a constant (i.e. `PROPERTY_CATEGORIES_LENGTH`)